pkg/binary/envoy/termination.go

```diff
@@ -15,40 +15,59 @@
 package envoy
 
 import (
+	"fmt"
+	"os"
 	"syscall"
 
+	"github.com/mholt/archiver/v3"
 	"github.com/tetratelabs/log"
 
 	"github.com/tetratelabs/getenvoy/pkg/binary"
 )
 
 func (r *Runtime) handleTermination() {
-	if r.cmd.Process == nil {
-		return // Envoy hasn't started at all
-	}
-	if r.cmd.ProcessState != nil {
-		if r.cmd.ProcessState.Success() {
-			log.Infof("Envoy process (PID=%d) exited successfully", r.cmd.Process.Pid)
-			return
-		}
-		log.Infof("Envoy process (PID=%d) terminated prematurely", r.cmd.Process.Pid)
-		return
-	}
+	cmd := r.cmd
+
+	defer interrupt(cmd.Process) // Ensure the SIGINT forwards to Envoy even if a pre-termination hook panics
 
+	log.Infof("GetEnvoy process (PID=%d) received SIGINT", os.Getpid())
 	// Execute all registered preTermination functions
 	for _, f := range r.preTermination {
 		if err := f(r); err != nil {
 			log.Error(err.Error())
 		}
 	}
 
-	// Forward on the SIGINT to Envoy
-	log.Infof("Sending Envoy process (PID=%d) SIGINT", r.cmd.Process.Pid)
-	r.cmd.Process.Signal(syscall.SIGINT) //nolint
+	interrupt(cmd.Process)
+}
+
+func interrupt(p *os.Process) {
+	log.Infof("Sending Envoy process (PID=%d) SIGINT", p.Pid)
+	_ = p.Signal(syscall.SIGINT)
+}
+
+func (r *Runtime) handlePostTermination() error {
+	for _, f := range r.postTermination {
+		if err := f(r); err != nil {
+			log.Error(err.Error())
```

Comment:
I recommend give context to this log, otherwise it is hard to know what was the application trying to do when this error happened.

```suggestion
			log.Errorf("failed to handle post termination: %v",  err)
```